src/ejabberd_gen_sm.erl

```diff
@@ -56,6 +61,13 @@ get_sessions(Mod, User, Server, Resource) ->
 create_session(Mod, User, Server, Resource, Session) ->
     Mod:create_session(User, Server, Resource, Session).
 
+-spec update_session(Mod :: module(), User :: jid:luser(),
```

Comment:
Because this is what we should use when talking to any database (persistent or not). Allowing the original value and then applying stringprep profiles is not the best approach IMO. Especially for internal modules like a backend.